### ISSUE TITLE
Differences between asset current balances and last value of charts of all transactions   

### ISSUE THREAD

**Issue by FrouxBY** [firefly-iii/firefly-iii#2438]
Good Evening, 

I am running Firefly III version 4.8.0.2, and my problem is:

I have almost  a year of data with 3 accounts
Account 1 : Euros, and only have euros transactions
Account 2 : Euros, with euros transactions + some exchange with a GBP account (account 3)
Account 3 : GBP, with GBP transactions + some exchange with an Euros account (account 2)

The current balance displayed on the assets view (firefly.example.org/accounts/asset) is no the same as in charts displayed either in the Dashboard or on in any assets views.
And the value change regarding of the date range chosen.

To be precised, the Account 1 is always correct (final amount when today included in the range = 
current balance)
And Account 2 and 3 got different values, which change depending the range.

**Steps to reproduce**
create 3 assets account, 2 in euros, 1 in GBP,
add some expenses and revenues in those 3 accounts,
Do some exchange between an euro asset and the GBP asset

**Expected behavior**
The final amount on chart should be coherent with the current balance, especially when displaying the "everything" range.

Here are some screenshot
![Annotation 2019-08-17 200125](https://user-images.githubusercontent.com/23535665/63216255-9723be00-c12a-11e9-969f-c73a45328a6e.jpg)
![Annotation 2019-08-17 200136](https://user-images.githubusercontent.com/23535665/63216256-97bc5480-c12a-11e9-8fbd-a0e09db5d413.jpg)
![Annotation 2019-08-17 200112](https://user-images.githubusercontent.com/23535665/63216254-9723be00-c12a-11e9-8905-8cb514c2b290.jpg)
![Annotation 2019-08-17 200049](https://user-images.githubusercontent.com/23535665/63216253-9723be00-c12a-11e9-87c4-49074b34cf18.jpg)

Last 30 days Chart :
![Annotation 2019-08-17 200400](https://user-images.githubusercontent.com/23535665/63216251-9723be00-c12a-11e9-9ccb-9e3f84747e4d.jpg)
Last 7 days Chart : 
![Annotation 2019-08-17 200453](https://user-images.githubusercontent.com/23535665/63216252-9723be00-c12a-11e9-9270-3c8dd1df880f.jpg)

Everything chart (from september 2018) : 
![Annotation 2019-08-17 200256](https://user-images.githubusercontent.com/23535665/63216257-97bc5480-c12a-11e9-95aa-e713326a97c7.jpg)

I had the same issues with Firefly 4.7 before the upgrade,

Thanks

**Comment by JC5** [comment]
Thanks. It's probably the exchange between the two accounts that's messing things up. I believe the chart and the balance thing use slightly different code. I'll check it out.

**Comment by JC5** [comment]
OK, this should be fixed in the next release. There is a routine that checks if transactions are stored correctly, BUT they might not catch up on the fix. Here's what you do after the next version is released.

On the command line, run:

```
php artisan firefly-iii:account-currencies --force
php artisan firefly-iii:other-currencies --force
```

If it's still broken, we'll need to dive deeper, but that will also mean I need some database exports to see which entries are broken (and how). We can pick it up in a new issue.

**Comment by FrouxBY** [comment]
Thank you very much, 
After upgrading, and running the two command line, 
I get 
`# php artisan firefly-iii:account-currencies --force
All accounts are OK.
Verified and fixed account currencies in 0.03 seconds.
# php artisan firefly-iii:other-currencies --force
Verified 728 transaction(s) and journal(s).
Verified and fixed transaction currencies in 1.13 seconds.`

And the numbers were still not correct, 
However, **now new transfert with different currency between account works correctly**, and only old one in the database are making the numbers wrong, deleting and recreating them fix everything (unfortunately, an update does not fix it). But as only transfert added between May and August were buggy, and not previous one, I could fix them all by hand. 

So finally everything is fine now, 
your doing great job !

**Comment by JC5** [comment]
Thanks!!
